libs/island-ui/contentful/src/lib/ProcessEntry/ProcessEntry.tsx

```diff
@@ -21,18 +26,61 @@ export const Titles: { [k: string]: string } = {
 export interface ProcessEntryProps {
   processTitle: string
   processLink: string
+  dropSignFileKey?: string
   type: string
   buttonText: string
 }
 
+export const openDocument = (
+  dropSignFileKey: string,
+  options?: OpenDocumentOptions,
+) => {
+  import('@taktikal/fillandsign').then(({ openDocument }) =>
+    Promise.resolve(openDocument(dropSignFileKey, options)),
+  )
+}
+
+export const getProcessEntryLinkProps = (
+  props: Pick<ProcessEntryProps, 'processLink' | 'dropSignFileKey'>,
+) => {
+  const { processLink, dropSignFileKey } = props
+
+  return {
+    ...(!dropSignFileKey && {
+      href: processLink,
+      icon: 'external' as IconProps['type'],
+    }),
+    ...(dropSignFileKey && {
+      onClick: () => openDocument(dropSignFileKey),
+    }),
+  }
+}
+
```

Comment:
This logic should probably be in it's own fill and sign component.
My concern is this might diverge from process entry both in design and functionality
e.g we might want to display an icon for fill and sign.

We might also use other similar services in future adding every service into this component might get complex.

Maybe the process entry component should take in props so we can wrap it with a fill and sign component?